### PR TITLE
bug fix, and add account_number (for IRA account)

### PR DESCRIPTION
**Pull request by mw66**
-- add check if instrument is None in get_option_market_data_by_id()
-- add account_number to get_all_open_option_orders()
-- add account_number to option_orders_url()

### CODE REVIEW

File: robin_stocks/robinhood/options.py [hunk 1]
```diff
@@ -334,43 +334,15 @@ def get_option_market_data_by_id(id, info=None):
...
-    url = marketdata_options_url()
-    payload = {
-        "instruments" : instrument['url']
-    }
-    data = request_get(url, 'results', payload)
-
-    if not data:
-        data= {
-        'adjusted_mark_price':'',
-        'ask_price':'',
```
Comment (mw66) [inline robin_stocks/robinhood/options.py:346]: e.g it makes no sense to check if data['ask_price'] is empty in the caller, and all other dict keys; instead the call should just check if data is None.

**Comment by farbodbahari** [comment]
Is there any update on this?

**Comment by mw66** [comment]
merged:  Jun 16, 2023

**Comment by farbodbahari** [comment]
> merged: Jun 16, 2023

is there any way to get the traded history and daily portfolio value of an IRA account? The 2 functions below for these 2 tasks only do it for the regular individual account by default and do not allow to specify account number.
robin.get_all_stock_orders() 
robin.get_historical_portfolio()

**Pull request merged**